server/routes/auth.js

```diff
@@ -1,18 +1,91 @@
 const express = require('express');
 const router = express.Router();
+const bcrypt = require('bcryptjs');
+const jwt = require('jsonwebtoken');
+const config = require('config');
+const auth = require('../middleware/auth');
+const {
+	body,
+	validationResult,
+} = require('express-validator');
+const User = require('../models/User');
 
 // @route   GET api/auth
 // @desc    Get logged in user
 // @access  Private
-router.get('/', (req, res) => {
-	res.send('Get logged in user');
+router.get('/', auth, (req, res) => {
+	try {
+		const user = await User.findById(req.user.id).select(
+			'-password'
+		);
+		res.json(user);
```

Comment:
Prob not worth, but did we need to verify the user id exists in the db? What if they were deleted?

Reply:
If there's no user, I think `select` will fail and the catch will run. Will have to check, otherwise yeah may as well check if exists.

Reply:
I was wrong, `select` doesn't trigger the catch. Added an if statement to check for and handle a null user value.